api/src/store.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+  "sort"
+  "time"
+)
+
+func LoadStore() (guilds []Guild, err error) {
+	// Open store.json file
+	file, err := os.OpenFile(STORE_FILE_NAME, os.O_CREATE, 0644)
+	defer file.Close()
+
+  if err != nil {
+    return guilds, err
+  }
+
+  // Read file into contents
+	contents, err := ioutil.ReadAll(file)
+	if err != nil {
+		return guilds, err
+	}
+
+  if len(contents) == 0 {
+    // Store is empty, no need to unmarshal
+    return guilds, err
+  }
+
+	err = json.Unmarshal(contents, &guilds)
+	return guilds, err
+}
+
+func (gs *GuildStore) WriteStore() (err error) {
+  gs.mutex.Lock()
+  defer gs.mutex.Unlock()
+
+  gs.Guilds = sortGuilds(gs.Guilds)
+
+  g, err := json.Marshal(gs.Guilds)
+  if err != nil {
+    return err
+  }
+
+  // Write marshalled guildStore.guilds to store.json
+  err = ioutil.WriteFile(STORE_FILE_NAME, g, 0644)
+  return err
+}
+
+func sortGuilds(guilds []Guild) []Guild {
+  sort.Slice(guilds, func(i, j int) bool {
+    return guilds[i].Timestamp > guilds[j].Timestamp
+  })
+
+  return guilds
+}
+
+func (gs *GuildStore) GuildInStore(guild Guild) (bool) {
```

Comment:
Add some comments to explain the difference between GuildStore and storedGuild

Reply:
This reminded me of the fact that since I do not work in Go, my approvals will mostly be based on how I could interact it with the api in my local setup. I'd suggest having someone who does with Go to approve future PR's.